pym/bob/cmds/build/builder.py

```diff
@@ -355,9 +356,11 @@ def setDownloadMode(self, mode):
             self.__archive.wantDownload(True)
             self.__downloadDepth = 0
             self.__downloadDepthForce = 1
-        else:
-            assert mode == 'no'
+        elif mode == 'no':
             self.__archive.wantDownload(False)
+        else:
+            self.__archive.wantDownload(True)
+            self.__downloadModules = re.compile(mode)
```

Comment:
We should catch a possible invalid regex here.

```suggestion
            try:
                self.__downloadModules = re.compile(mode)
            except re.error as e:
                raise BuildError("Invalid download regex '{}': {}".format(e.pattern), e)
```